ui/components/src/model/legend.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { MouseEventHandler } from 'react';
+
+export const legendPositions = ['Bottom', 'Right'] as const;
+
+export type LegendPositions = typeof legendPositions[number];
```

Comment:
is it simpler if we just do `type LegendPositions = 'Bottom' | 'Right'`?

Reply:
So I ended up with it this way so I could use legendPositions further down in isValidLegendPosition: https://github.com/perses/perses/blob/d8cf60b09b5965eeb17773706f25efd390829ad8/ui/components/src/model/legend.ts#L55